include/sys/dmu_objset.h

```diff
@@ -217,6 +217,7 @@ boolean_t dmu_objset_userobjused_enabled(objset_t *os);
 boolean_t dmu_objset_userobjspace_upgradable(objset_t *os);
 void dmu_objset_userobjspace_upgrade(objset_t *os);
 boolean_t dmu_objset_userobjspace_present(objset_t *os);
+boolean_t dmu_objset_bad_encryption_version(objset_t *os);
```

Comment:
Rather than `dmu_objset_bad_encryption_version()`, which I admit is descriptive, let's make the generic interface return the version number even if we don't expect to have to bump this version again.  A convenience wrapper function can be added to check if that version is compatible, or not, and include an explanation why.  Calling it bad is misleading.